src/resonate/functools.py

```diff
@@ -78,3 +79,22 @@ def wrap_fn(
         assert iscoroutinefunction(fn)
         cmd = AsyncFnWrapper(ctx, fn, *args, **kwargs)
     return cmd
+
+
+def run_with_retry_policy(
```

Comment:
I am might be missing it, but I couldn't find a place where calculate_delay is used, which I think causes the retries to not wait in between attempts